client/tokenized-express-checkout/transformers/wc-to-stripe.js

```diff
@@ -78,25 +86,71 @@ export const transformCartDataForDisplayItems = ( rawCartData ) => {
 			.join( ' ' ),
 	} ) );
 
-	const taxAmount = parseInt( cartData.totals.total_tax || '0', 10 );
-	if ( taxAmount ) {
-		displayItems.push( {
-			amount: transformPrice( taxAmount, cartData.totals ),
-			name: __( 'Tax', 'woocommerce-payments' ),
-		} );
-	}
-
 	const shippingAmount = parseInt(
 		cartData.totals.total_shipping || '0',
 		10
 	);
 	if ( shippingAmount ) {
 		displayItems.push( {
-			amount: transformPrice( shippingAmount, cartData.totals ),
+			amount: transformPrice(
+				displayPriceIncludingTax
+					? shippingAmount +
+							parseInt(
+								cartData.totals.total_shipping_tax || '0',
+								10
+							)
+					: shippingAmount,
+				cartData.totals
+			),
```

Comment:
I initially thought of having a utility function for this. But then, I noticed that WC Core performs the same check in multiple areas, so I thought it might not be necessary 🤷 

https://github.com/woocommerce/woocommerce/blob/f4d80351eda1dab106db285827cf99ae5432103f/plugins/woocommerce/client/blocks/assets/js/blocks/checkout/inner-blocks/checkout-shipping-methods-block/block.tsx#L39-L41

https://github.com/woocommerce/woocommerce/blob/f4d80351eda1dab106db285827cf99ae5432103f/plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/shipping-rates-control-package/render-package-rate-option.tsx#L20-L25